src/cli/managers/config_manager.py

```diff
@@ -18,8 +21,33 @@ def get_application(self, application: str) -> Dict[str, str]:
     def get_application_from_path(self, path: Path) -> Tuple[str, Dict[str, str]]:
         return "key", {}
 
-    def get_applications(self) -> List[Dict[str, str]]:
-        return [{}]
+    def get_applications(self) -> List[Dict[str, Any]]:
+        """
+        Reads the applications.json config file for getting the local applications
+
+        Returns:
+            A list of applications available in the config file
+
+        Raises:
+            MalformedJsonFile: If the config file contains invalid json
+        """
+        applications_config = self.__config_dir / 'applications.json'
+        application_list = []
+        if applications_config.is_file():
+            try:
+                with open(applications_config, encoding="utf-8") as fp:
+                    applications = json.load(fp)
+                    for app_name, app_data in applications.items():
+                        app_data['name'] = app_name
+                        application_list.append(app_data)
+                    return application_list
+            except json.decoder.JSONDecodeError as exception:
+                logging.error('The file %s does not contain valid json. Error: %s', applications_config, exception)
+                raise MalformedJsonFile(exception) from exception
+        else:
+            logging.info('The configuration file %s was not found. '
```

Comment:
Configuration file will be created when the user creates their first local application. 
If the user runs the application list command before creating any local application, i think it should not throw any error. It can just let the user know that there are no applications.

Reply:
Exactly what I meant. It is not an error to do a list before you created one application. Agree with only informing user (no apps)